src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/Ordering/HACK_AddItemHelper.cs

```diff
@@ -67,16 +67,16 @@ private enum AddItemAction { NewItem = 0, ExistingItem = 1 }
         /// </summary>
         private static int ShowAddItemDialog(SVsServiceProvider serviceProvider, IProjectTree target, IVsProject vsProject, string strBrowseLocations, AddItemAction addItemAction)
         {
-            var addItemDialog = serviceProvider.GetService<IVsAddProjectItemDlg, SVsAddProjectItemDlg>();
+            IVsAddProjectItemDlg addItemDialog = serviceProvider.GetService<IVsAddProjectItemDlg, SVsAddProjectItemDlg>();
             Assumes.Present(addItemDialog);
 
-            var uiFlags = __VSADDITEMFLAGS.VSADDITEM_AddNewItems | __VSADDITEMFLAGS.VSADDITEM_SuggestTemplateName | __VSADDITEMFLAGS.VSADDITEM_AllowHiddenTreeView;
+            __VSADDITEMFLAGS uiFlags = __VSADDITEMFLAGS.VSADDITEM_AddNewItems | __VSADDITEMFLAGS.VSADDITEM_SuggestTemplateName | __VSADDITEMFLAGS.VSADDITEM_AllowHiddenTreeView;
             if (addItemAction == AddItemAction.ExistingItem)
             {
                 uiFlags = __VSADDITEMFLAGS.VSADDITEM_AddExistingItems | __VSADDITEMFLAGS.VSADDITEM_AllowMultiSelect | __VSADDITEMFLAGS.VSADDITEM_AllowStickyFilter | __VSADDITEMFLAGS.VSADDITEM_ProjectHandlesLinks;
             }
 
-            var strFilter = string.Empty;
+            string strFilter = string.Empty;
```

Comment:
just my personal preference.. however don't you think that this is very redundant? For example, look at all changes you did to this file - they are so obvious what type is there and it takes so much time to type all those explicit types .. 

why you need explicit string x = string.Empty? When i think about explicitly typing CPS' async event handlers parameters types , i really "scared"  :) ..

Reply:
This is: https://github.com/dotnet/roslyn/issues/23714.